mm2src/coins/z_coin.rs

```diff
@@ -394,18 +401,128 @@ impl ZCoin {
         let total_output = big_decimal_from_sat_unsigned(total_output_sat, self.utxo_arc.decimals);
         let total_required = &total_output + &tx_fee;
 
-        let spendable_notes = self
-            .spendable_notes_ordered()
-            .await
-            .map_err(|err| GenTxError::SpendableNotesError(err.to_string()))?;
+        let mut retries = 0;
+
+        loop {
+            // Wait for block sync and previous tx confirmation.
+            let sync_guard = self.wait_for_gen_tx_blockchain_sync().await?;
+            let spendable_notes = self
+                .spendable_notes_ordered()
+                .await
+                .map_err(|err| GenTxError::SpendableNotesError(err.to_string()))?;
+
+            let change_notes = self.z_fields.locked_notes_db.load_all_notes().await?;
+            println!("LOCKED NOTES: {change_notes:?}");
```

Comment:
Shouldn't it be info! instead of println?

Reply:
I didn't mean to commit this..removed